libraries/AP_HAL_ChibiOS/hwdef/F4BY_F427/hwdef.dat

```diff
@@ -0,0 +1,177 @@
+
+# USART2 serial2 telem2
+PD5 USART2_TX USART2 NODMA
+PD6 USART2_RX USART2
+
+# USART3 serial3 telem1
+PD8 USART3_TX USART3 NODMA
+PD9 USART3_RX USART3
+
+PC10 UART4_TX UART4 NODMA
+PC11 UART4_RX UART4
+
+# SHARE dma with I2C2_TX
+PC12 UART5_TX UART5 NODMA
+PD2  UART5_RX UART5 NODMA
+
+
+#SPI1 for MPU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+PA4 MPU_CS CS
+
+# spi bus for dataflash AND SD
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+
+PB12 FRAM_CS CS SPEED_VERYLOW
+PE15 FLASH_CS CS
+
+SPIDEV mpu6000 SPI1 DEVID1 MPU_CS   MODE3 1*MHZ   8*MHZ
+SPIDEV ramtron SPI2 DEVID2 FRAM_CS  MODE3 8*MHZ   8*MHZ
+SPIDEV sdcard  SPI2 DEVID3 FLASH_CS MODE0 400*KHZ 25*MHZ
+
+# one IMU
+IMU Invensense SPI:mpu6000 ROTATION_NONE
+
+# one baro, check both addresses
+BARO MS56XX I2C:0:0x76
+BARO MS56XX I2C:0:0x77
+
+# enable RAMTROM parameter storage
+define HAL_WITH_RAMTRON 1
+# enable FAT filesystem support
+define HAL_OS_FATFS_IO 1
+
+# this defines the default maximum clock on I2C devices.
+define HAL_I2C_MAX_CLOCK 100000
+I2C_ORDER I2C2 I2C1
+PB8 I2C1_SCL I2C1
+PB9 I2C1_SDA I2C1
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# look for I2C compass
+COMPASS HMC5843 I2C:0:0x1E false ROTATION_YAW_270
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+# PWM out pins
+PA0 TIM2_CH1 TIM2 PWM(1) GPIO(50)
+PA1 TIM2_CH2 TIM2 PWM(2) GPIO(51)
+PA2 TIM2_CH3 TIM2 PWM(3) GPIO(52)
+PA3 TIM2_CH4 TIM2 PWM(4) GPIO(53)
+PE9 TIM1_CH1 TIM1 PWM(5) GPIO(54)
+PE11 TIM1_CH2 TIM1 PWM(6) GPIO(55)
+PE13 TIM1_CH3 TIM1 PWM(7) GPIO(56)
+PE14 TIM1_CH4 TIM1 PWM(8) GPIO(57)
+
+PD13 TIM4_CH2 TIM4 PWM(9) GPIO(58)
+PD12 TIM4_CH1 TIM4 PWM(10) GPIO(59)
+PD15 TIM4_CH4 TIM4 PWM(11) GPIO(60)
+PD14 TIM4_CH3 TIM4 PWM(12) GPIO(61)
+
+# also USART6_RX for unidirectional RC,including PPM
+PC7 TIM8_CH2 TIM8 RCININT PULLDOWN LOW DMA_CH0
+# PC6 TIM8_CH1 TIM8 RCININT PULLDOWN LOW DMA_CH0
+
+
+# New style Pixracer LED configuration for master repo
+define AP_NOTIFY_GPIO_LED_RGB_ENABLED 1
+
+PE3 LED_RED   OUTPUT GPIO(10)
+PE2 LED_GREEN OUTPUT GPIO(11)
+PE1 LED_BLUE  OUTPUT GPIO(12)
+PE0 LED_YELOW OUTPUT GPIO(13)
+
+define AP_NOTIFY_GPIO_LED_RGB_RED_PIN 10
+define AP_NOTIFY_GPIO_LED_RGB_GREEN_PIN 11
+define AP_NOTIFY_GPIO_LED_RGB_BLUE_PIN 12
+
+
+
+PC0 PRESSURE_SENS ADC1 SCALE(1)
+PC1 RSSI_IN ADC1
+PC2 BATT_CURRENT_SENS ADC1 SCALE(2)
+PC3 BATT_VOLTAGE_SENS ADC1 SCALE(2)
+PC4 VDD_5V_SENS ADC1 SCALE(2)
+PC5 FMU_SERVORAIL_VCC_SENS ADC1 SCALE(2)
+
+
+PE5 TIM9_CH1 TIM9 ALARM
+PC14 EXTERN_GPIO1 OUTPUT GPIO(1)
+PC13 EXTERN_GPIO2 OUTPUT GPIO(2)
+PE4  EXTERN_GPIO3 OUTPUT GPIO(3)
+PE6  EXTERN_GPIO4 OUTPUT GPIO(4)
+PC9  EXTERN_GPIO5 OUTPUT GPIO(5)
+
+#  IRQ for MPU6000
+PB0 EXTI_MPU6000 INPUT PULLUP
+PB1 DRDY_HMC5883 INPUT PULLUP
+
+# this constants with HAL_
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 13
+define HAL_BATT_CURR_PIN 12
+define HAL_BATT_VOLT_SCALE 16.04981
+define HAL_BATT_CURR_SCALE 100
+
+# this constant with AP_
+define AP_BATT_CURR_AMP_OFFSET_DEFAULT 2.316
```

Comment:
wow! that is a huge offset! most need none or something very small.....this is suspicious

Reply:
> wow! that is a huge offset! most need none or something very small.....this is suspicious

We use a bipolar sensor based on a hall effect Allegro ACS758. It measures both discharge and charge. thus its midpoint is about half the voltage